src/components/MdComponents/index.tsx

```diff
@@ -0,0 +1,220 @@
+import { ComponentProps, type HTMLAttributes } from "react"
+import { Badge, Box, type BoxProps } from "@chakra-ui/react"
+
+import type { ChildOnlyProp } from "@/lib/types"
+
+import ButtonDropdown, {
+  type ButtonDropdownProps,
+} from "@/components/ButtonDropdown"
+import Contributors from "@/components/Contributors"
+import MarkdownImage from "@/components/MarkdownImage"
+import { mdxTableComponents } from "@/components/Table"
+import TooltipLink from "@/components/TooltipLink"
+import YouTube from "@/components/YouTube"
+
+import { cn } from "@/lib/utils/cn"
+
+import ContributorsQuizBanner from "../Banners/ContributorsQuizBanner"
+import Card from "../Card"
+import DocLink from "../DocLink"
+import Emoji from "../Emoji"
+import ExpandableCard from "../ExpandableCard"
+import FeaturedText from "../FeaturedText"
+import GlossaryTooltip from "../Glossary/GlossaryTooltip"
+import IdAnchor from "../IdAnchor"
+import InfoBanner from "../InfoBanner"
+import IssuesList from "../IssuesList"
+import LocaleDateTime from "../LocaleDateTime"
+import MainArticle from "../MainArticle"
+import { StandaloneQuizWidget } from "../Quiz/QuizWidget"
+import { ButtonLink } from "../ui/buttons/Button"
+import { Flex } from "../ui/flex"
+import { ListItem, OrderedList, UnorderedList } from "../ui/list"
+
+export const commonHeadingAttributes = (className: string, id?: string) => ({
+  id,
+  className: cn(
+    "font-bold leading-xs my-8",
+    id && "scroll-mt-28 relative",
+    className
+  ),
+  "data-group": !!id || undefined,
```

Comment:
@pettinarip I'm not sure here, actually. And I actually don't know what this attribute is here at all. It doesn't look like it's needed for anything.

Reply:
Right, that was for Chakra purposes I think, no? 🤔

Reply:
Yeah, this is related to the `IdAnchor` component to display the anchor icon on hover.

Reply:
@pettinarip ahh I remember now. You cannot allow `data-group` to render a `false` value. Because the hover styling if `data-group` is present, regardless if it's value is `true`. 

So passing `true` or `undefined` ensures that if not `true` then the attribute is not rendered, thereby preventing the hover styles.

Reply:
Ahhh makes sense lol tricky one